pyvista/plotting/lights.py

```diff
@@ -0,0 +1,466 @@
+        respectively. The class constants ``Light.HEADLIGHT``, ``Light.CAMERA_LIGHT``
+        and ``Light.SCENE_LIGHT`` are also available, respectively.
+
+        A headlight is attached to the camera, looking at its focal point along
+        the axis of the camera.
+        A camera light also moves with the camera, but it can occupy a general
+        position with respect to it.
+        A scene light is stationary with respect to the scene, as it does not
+        follow the camera. This is the default.
+    """
+
+    # TODO: better/more explanation for ``position``?
+
+    # pull in light type enum values as class constants
+    HEADLIGHT = LightType.HEADLIGHT
+    CAMERA_LIGHT = LightType.CAMERA_LIGHT
+    SCENE_LIGHT = LightType.SCENE_LIGHT
+    # TODO: is this design OK?
+
+    def __init__(self, position=None, color=None, light_type='scene light'):
+        """Initialize the light."""
+        super().__init__()
+
+        if position:
+            self.position = position
+
+        if color:
+            self.ambient_color = color
+            self.diffuse_color = color
+            self.specular_color = color
+            # TODO: make this a single call if that's implemented
+
+        if isinstance(light_type, str):
+            # be forgiving: ignore spaces and case
+            light_type_orig = light_type
+            type_normalized = light_type.replace(' ', '').lower()
+            mapping = {'headlight': LightType.HEADLIGHT,
+                       'cameralight': LightType.CAMERA_LIGHT,
+                       'scenelight': LightType.SCENE_LIGHT,
+                      }
+            try:
+                light_type = mapping[type_normalized]
+            except KeyError:
+                raise ValueError(f'Invalid ``light_type`` "{light_type_orig}"') from None
+        elif not isinstance(light_type, int):
+            raise TypeError('Parameter ``light_type`` must be int or str,'
+                            f' not {type(light_type)}.')
+        # LightType is an int subclass
+
+        self.light_type = light_type
+
+        # TODO: more init kwargs (focal_point, cone angle etc.)
+        # TODO: default Positional state?
+        # TODO: should setting attenuation and cone angle automatically switch to positional?
+        # TODO: should color and point and direction_angle have more flexible signatures? (only for non-properties)
+        # TODO: ndarray type and shape and size checking for color and point
+
+    def __repr__(self):
+        """Print a repr specifying the id of the light and its light type."""
+        return (f'<{self.__class__.__name__} ({self.light_type}) at {hex(id(self))}>')
+
+    #### Properties ####
+
+    @property
+    def ambient_color(self):
+        """Return the ambient color of the light."""
+        return self.GetAmbientColor()
+
+    @ambient_color.setter
+    def ambient_color(self, color):
+        """Set the ambient color of the light."""
+        self.SetAmbientColor(color)
```

Comment:
Here and elsewhere, we need examples within the docstrings for the properties (not necessarily the setter).  I'll add a sphinx autodoc for this class and it will be helpful to have examples for these methods/attributes.

Additionally, `color` should accept a string as in our `plotter` API using `pyvista.plotting.theme.parse_color`.

Reply:
Is my understanding correct that setter docstrings don't show up anywhere, and getter docstrings show up in `help(light)`? Hmm, and I guess sphinx might use these too, I'll look that up in the existing docs.

And thanks for the color string suggestion, good catch.